tests/unit/test_snowflake_adapter.py

```diff
@@ -317,7 +317,7 @@ def test_client_session_keep_alive_true(self):
                     application="dbt",
                     insecure_mode=False,
                     session_parameters={},
-                    reuse_connections=None,
```

Comment:
Hm, so above I see that you set this logic to say "if reuse connections is None and client_session_keep_alive is True, leave it". Shouldn't this stay `None` then?

What's are these tests actually testing if not what configs are allowed / unallowed

Reply:
Yes, you're correct. I'm surprised this unit test passed, good catch.

Reply:
Ah, it's because the `Credentials` class is initially created with `client_session_keep_alive=False`, hence the `__post_init__` sets the default for `reuse_connections`. Then `replace` is used to replace the property `client_session_keep_alive` which does not retrigger `__post_init__`.

Reply:
I updated the test to set both, with an explanation as to why `reuse_connections` needs to be reset. I don't love it, but I don't know another way to test it without re-configuring this test class to not use `replace`. I'm open to suggestions.